final class androidx.concurrent.futures.AbstractResolvableFuture$Failure extends java.lang.Object
{
    static final androidx.concurrent.futures.AbstractResolvableFuture$Failure FALLBACK_INSTANCE;
    final java.lang.Throwable exception;

    static void <clinit>()
    {
        androidx.concurrent.futures.AbstractResolvableFuture$Failure $r0;
        androidx.concurrent.futures.AbstractResolvableFuture$Failure$1 $r1;

        $r0 = new androidx.concurrent.futures.AbstractResolvableFuture$Failure;

        $r1 = new androidx.concurrent.futures.AbstractResolvableFuture$Failure$1;

        specialinvoke $r1.<init>("Failure occurred while trying to finish a future.");

        specialinvoke $r0.<init>($r1);

        androidx.concurrent.futures.AbstractResolvableFuture$Failure.FALLBACK_INSTANCE = $r0;

        return;
    }

    void <init>(java.lang.Throwable)
    {
        java.lang.Throwable $r1;
        java.lang.Object $r2;
        androidx.concurrent.futures.AbstractResolvableFuture$Failure r0;

        r0 := @this;

        $r1 := @parameter0;

        specialinvoke r0.<init>();

        $r2 = androidx.concurrent.futures.AbstractResolvableFuture.checkNotNull($r1);

        $r1 = (java.lang.Throwable) $r2;

        r0.exception = $r1;

        return;
    }
}
